backpack/core/derivatives/adaptive_avg_pool_nd.py

```diff
@@ -0,0 +1,105 @@
+"""Implements the derivatives for AdaptiveAvgPool."""
+from typing import List, Tuple, Union
+from warnings import warn
+
+from torch import Size
+from torch.nn import AdaptiveAvgPool1d, AdaptiveAvgPool2d, AdaptiveAvgPool3d
+
+from backpack.core.derivatives.avgpoolnd import AvgPoolNDDerivatives
+
+
+class AdaptiveAvgPoolNDDerivatives(AvgPoolNDDerivatives):
+    """Implements the derivatives for AdaptiveAvgPool."""
+
+    def check_parameters(
+        self, module: Union[AdaptiveAvgPool1d, AdaptiveAvgPool2d, AdaptiveAvgPool3d]
+    ) -> None:
+        """Checks if the parameters are supported.
+
+        Specifically checks if input shape is multiple of output shape.
+        In this case, there are parameters for AvgPoolND that are equivalent.
+
+        https://stackoverflow.com/questions/53841509/how-does-adaptive-pooling-in-pytorch-work/63603993#63603993 # noqa: B950
+
+        Args:
+            module: module to check
+
+        Raises:
+            NotImplementedError: if the given shapes do not match
+        """
+        if ("cuda" in str(module.input0.device)) and (self.N == 3):
+            warn(
+                "Be careful when computing gradients of AdaptiveAvgPool3d. "
+                "There is a bug using autograd.grad on cuda with AdaptiveAvgPool3d. "
+                "https://discuss.pytorch.org/t/bug-report-autograd-grad-adaptiveavgpool3d-cuda/124614 "  # noqa: B950
+                "BackPACK derivatives are correct."
+            )
+
+        shape_input: Size = module.input0.shape
+        shape_output: Size = module.output.shape
+
+        # check length of input shape
+        if not len(shape_input) == (self.N + 2):
+            raise NotImplementedError(
+                f"input must be (batch_size, C, ...) with ... {self.N} dimensions"
+            )
+
+        # check if input shape is multiple of output shape
+        if any([shape_input[2 + n] % shape_output[2 + n] != 0 for n in range(self.N)]):
+            raise NotImplementedError(
+                f"No equivalent AvgPool (unadaptive): Input shape ({shape_input}) "
+                f"must be multiple of output shape ({shape_output})."
+            )
+
+    def get_parameters(
+        self, module: Union[AdaptiveAvgPool1d, AdaptiveAvgPool2d, AdaptiveAvgPool3d]
+    ) -> Tuple[List[int], List[int], List[int]]:
+        """Return parameters for an equivalent AvgPool.
```

Comment:
Discussion: Can this naming convention cause confusion? Module parameters could be misinterpreted as the items of `module.parameters()`. Would changing `parameters` to `hyperparameters` resolve such a misunderstanding?

Reply:
Yes, I agree that it can cause confusion, especially for third party users.
`check_parameters` -> `check_module_parameters`
`get_parameters` -> `get_avg_pool_equivalent_parameters`
I admit that these names are lengthy but I think they avoid any misunderstanding. If misunderstanding is still possible, then `hyperparameters` is better

Reply:
Do you agree to do the following?
- leave `check_parameters` for now
- change `get_parameters` to `get_avg_pool_equivalent_parameters`

Reply:
Agree. Linking to https://github.com/fKunstner/backpack-discuss/issues/105

Reply:
Changed `get_parameters` to `get_avg_pool_parameters`.
Because this method is shared with parent class `AvgPoolNDDerivatives` the name `get_avg_pool_equivalent_parameters` doesn't make sense.